### PR TITLE
Plaid bar stacked

### PR DESCRIPTION
**Pull request by kturney**
on top of https://github.com/ivanvanderbyl/maximum-plaid/pull/24

### CODE REVIEW

File: addon/components/plaid-bar-stacked/component.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      y = (d) => yScale(d[1]);
+      height = (d) => yScale(d[0]) - yScale(d[1]);
+    } else {
+      x = (d) => xScale(d[0]);
+      width = (d) => xScale(d[1]) - xScale(d[0]);
+      y = (d) => yScale(d.data[stackBy]);
+      height = yScale.bandwidth;
+    }
+
+    let appendElements = values !== this.drawnValues || values.length !== this.drawnValues.length;
```
Comment (ivanvanderbyl) [inline addon/components/plaid-bar-stacked/component.js]: I don't think we need this. Just use the d3 update/merge/remove pattern.

**Comment by ivanvanderbyl** [comment]
A few thoughts on how stacked charts should work:
1. The chart shouldn't do any data processing, this should all be done with helpers/controllers etc. Which means the chart should receive a defined data set which makes sense for stacking. I think this would probably be some sort of stacked helper (like the one you built) which outputs a stack.
2. Each bar group can be stacked vertically or side by side in groups. There's very little difference in implementation, just the positioning of the `<rect>`s.

**Comment by kturney** [comment]
@ivanvanderbyl do we want area and bar to just handle it if they receive stacked data, or does it make sense to have a stacked-area and stacked-bar?

**Comment by ivanvanderbyl** [comment]
@kturney I think it makes sense to have each automatically handle it. We already expect data to be in one format, perhaps we should also specify a stacked format which we can detect from a stacked-layout helper:

``` hbs
  {{plot.bar (stacked-layout data)}}
```

**Pull request closed without merging**